pyocd/probe/stlink_probe.py

```diff
@@ -355,8 +355,8 @@ def write_memory_block8(self, addr: int, data: Sequence[int], **attrs: Any) -> N
         idx = 0
 
         # write leading unaligned bytes
-        unaligned_count = addr & 3
-        if (size > 0) and (unaligned_count > 0):
+        unaligned_count = 3 & (4 - (addr & 3))
+        if (size > unaligned_count > 0):
```

Comment:
Same as above.

Reply:
What... if size 1 and unaligned_count 2?
if (1> 2 > 0):
      # not exe

Reply:
This section should have the same change as requested above:

```py
unaligned_count = min(size, 3 & (4 - addr))
if unaligned_count > 0:
```